scala/src/main/scala/smile/clustering/k-modes/BinaryClusteringTypes.scala

```diff
@@ -0,0 +1,9 @@
+package smile.clustering.clusteringTypes
+
+trait BinaryClusteringTypes
```

Comment:
I don't see the necessary to define this trait. If you have to define these types, just define them in the `KModes` class. In fact, I don't feel these types are necessary.